src/kkos2-display-bundle/ExternalData/EventfeedHelper.php

```diff
@@ -53,32 +53,58 @@ public function setSlide(SlidesInSlide $slide, string $slideType) {
   }
 
   /**
-   * @param string $url Url to feed
-   * @param integer $numItems Number of items to return from the feed
-   * @param array $queryData
+   * Fetch data from feeds.
    *
-   * @return array|mixed
+   * Feeds are provieded via the "datafeed_url" option on the slide, it contains
+   * zero or more feed urls separated by newlines.
+   *
+   * @return array
+   *   An array of json objects.
    */
   public function fetchData() {
-    $url = $this->slide->getOption('datafeed_url', '');
-    if (!$this->validateFeedUrl()) {
-      throw new \Exception("$url is not a valid {$this->slideType} url.");
+    $datafeed_urls = $this->slide->getOption('datafeed_url', '');
+    if (empty($datafeed_urls)) {
+      return [];
     }
-    $query = [];
-    $filterDisplay = $this->slide->getOption('datafeed_display', '');
-    if (!empty($filterDisplay)) {
-      $query = [
-        'display' => $filterDisplay,
-      ];
+
+    // Assume we get multiple feeds.
+    $urls = array_map(
+      "trim",
+      explode("\n", $datafeed_urls)
+    );
```

Comment:
Have you considered using [`preg_split(\s*\n\s*)`](https://www.php.net/manual/en/function.preg-split.php) instead?

I like a little functional programming but it does not seem to be used that much here. With `preg_split()` we can use modifiers like `PREG_SPLIT_NO_EMPTY` and avoid potential problems with empty lines.

Reply:
I'll add a `array_filter` to get rid of empty entries, and drop the nested calls to make the code more clear.

I get your point about functional programming, I'm switching between languages quite a bit these days so I guess I'm influenced by javascript in this case.  With that being said I don't find the reg-ex approach more clear. My thinking went like this when I looked at the preg_split "but... \s matches newlines as well, so why is in there? A thats right, its normally greedy but as the \n is in there, it will only be able to do a full match if it actually finds a newline. Aaaand with the sourrounding \s* we get an implicit trim - got it" - adding in PREG_SPLIT_NO_EMPTY would probably also require an additional mental cycle or two.

It's interesseting - In Go you would mumble something about "idiomatic" and then require the reader to recognise a pattern here and just know that this is the classic "preg-split with a trim" pattern.

So all in all, good observation! I'm going with an improved version of "my" approach, all while thinking that maybe someone should do a session or two someday on idiomatic php just for the fun of it :D